### ISSUE TITLE
Feature request: Automatic handling of compressed images

### ISSUE THREAD

**Issue by emk2203** [jsamr/bootiso#55]
It would be nice to have Etcher-like functionality to detect and handle compressed images like img.xz.

This in addition to handling img files in general.

**Comment by jsamr** [comment]
@emk2203 The thing is, it would add a lot of new dependencies for what seems like an edge-case. What is your typical use case?

**Comment by emk2203** [comment]
The typical use case would be just that: Taking a downloaded system image in `img.xz` format and having bootiso handle it transparently. These are used a lot for Raspberry Pi and other single-board computers. When using `dd`, I would just pipe the file from the decompressor, but with `bootiso`, it's not so easy.

The process would be:

* feed file into bootiso when prompted
   * bootiso uses as-is if img or iso file
   * with .xz extension, file gets passed through `xz`first for on-the-fly decompression

If `xz` is not installed, which is unlikely, just an error could be given.

**Comment by davidebeatrici** [comment]
For reference:

`xzcat file.img.xz | dd of=/dev/sdx`

**Comment by jsamr** [comment]
> For reference:
> 
> `xzcat file.img.xz | dd of=/dev/sdx`

We don't use dd 100% of the time. For instance, to handle windows images or when we manually install a bootloader.

**Comment by davidebeatrici** [comment]
Right. I believe supporting `xz` archives for Linux images would already be a great start though.
